### PR TITLE
Embed: validate query arguments

### PR DESCRIPTION
**Pull request by stsewd**
- Validate that the correct arguments are given
- Don't return http requests from the parser
- Always update doc from path (in case path is given by the url)
- Hit the API to test things

### CODE REVIEW

File: readthedocs/embed/views.py [hunk 1]
```diff
@@ -89,23 +89,21 @@ class EmbedAPIBase(APIView):
...
-        * url (required)
+    * url (with fragment) (required)
 
     ### Example
 
-        GET https://readthedocs.org/api/v2/embed/?project=requests&doc=index&section=User%20Guide&path=/index.html  # noqa
-
-        GET https://readthedocs.org/api/v2/embed/?url=https://docs.readthedocs.io/en/latest/features.html%23github-bitbucket-and-gitlab-integration # noqa
+    - GET https://readthedocs.org/api/v2/embed/?project=requestsF&version=latest&doc=index&section=User%20Guide&path=/index.html  # noqa
+    - GET https://readthedocs.org/api/v2/embed/?url=https://docs.readthedocs.io/en/latest/features.html%23github-bitbucket-and-gitlab-integration # noqa
```
Comment (ericholscher) [inline readthedocs/embed/views.py:106]: Does this #noqa show up in the docs? We should figure out a better way to ignore that, if so.

File: readthedocs/embed/tests/test_api.py [hunk 2]
```diff
@@ -52,8 +57,49 @@ def _get_html_content(self, html_file):
...
+            'url=http://project.readthedocs.io/en/latest/#title-one',
+            'url=http://project.readthedocs.io/en/latest/index.html?foo=bar#title-one',
+            'url=http://project.readthedocs.io/en/latest/?foo=bar#title-one',
+
+            # doc & path
+            f'project={self.project.slug}&version={self.version.slug}&path=index.html&section=title-one',
+            f'project={self.project.slug}&version={self.version.slug}&path=/index.html&section=title-one',
+            f'project={self.project.slug}&version={self.version.slug}&doc=index&section=title-one',
+            f'project={self.project.slug}&version={self.version.slug}&path=index.html&doc=index&section=title-one',
+        )
```
Comment (ericholscher) [inline readthedocs/embed/tests/test_api.py:95]: Might be nicer to construct these from kwargs and url encode them, so it's easier to tell what we're changing between URL's. Maybe that's too much work tho.

**Review by ericholscher — APPROVED** [review]
Having these test more of the actual serving code is 💯

File: readthedocs/embed/views.py (hunk 1 above)
Comment (stsewd) [reply readthedocs/embed/views.py]: Yeah it does, I was able to disable the linter here with pylint ignore and #noqa

File: readthedocs/embed/tests/test_api.py (hunk 2 above)
Comment (stsewd) [reply readthedocs/embed/tests/test_api.py:123]: turns our the test client accepts a dic for the query strings of get

**Pull request merged**